File: src/contests/cg/puzzle/scala/the-river-i-.scala
// 🎮 CodinGame Puzzle - the-river-i-
// https://www.codingame.com/training/easy/the-river-i-

object Solution {
  def digitSum(x0: Long): Long = {
    var x = x0
    var s = 0L
    while (x > 0) {
      s += x % 10
      x /= 10
    }
    s
  }

  def main(args: Array[String]): Unit = {
    var a = scala.io.StdIn.readLine().trim.toLong
    var b = scala.io.StdIn.readLine().trim.toLong
    while (a != b) {
      if (a < b) a += digitSum(a) else b += digitSum(b)
    }
    println(a)
  }
}

File: src/contests/cg/puzzle/csharp/the-river-i-.cs
// 🎮 CodinGame Puzzle - the-river-i-
// https://www.codingame.com/training/easy/the-river-i-

using System;

class Solution
{
    static long DigitSum(long x)
    {
        long s = 0;
        while (x > 0)
        {
            s += x % 10;
            x /= 10;
        }
        return s;
    }

    static void Main()
    {
        long a = long.Parse(Console.ReadLine().Trim());
        long b = long.Parse(Console.ReadLine().Trim());
        while (a != b)
        {
            if (a < b) a += DigitSum(a); else b += DigitSum(b);
        }
        Console.WriteLine(a);
    }
}
